ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -141,6 +144,15 @@ public Object processInvocation(InterceptorContext invocation) throws Exception
                 case NOT_SUPPORTED:
                     return notSupported(invocation, component);
                 case REQUIRED:
+                    final ComponentView view = invocation.getPrivateData(ComponentView.class);
+                    if (view != null && view.isAsynchronous(method)) {
+                        // EJB 3.2 4.5.3 Transactions
+                        // The client’s transaction context does not propagate with an asynchronous method invocation. From the
+                        // Bean Provider’s point of view, there is never a transaction context flowing in from the client. This
+                        // means, for example, that the semantics of the REQUIRED transaction attribute on an asynchronous
+                        // method are exactly the same as REQUIRES_NEW.
+                        return requiresNew(invocation, component, timeoutInSeconds);
```

Comment:
How did we get to the point of having a tx here during an async invocation?

Reply:
Not sure why, it could be some missed check during earlier part of invocation chain. I'll look into it.